src/H5Dchunk.c

```diff
@@ -3241,6 +3255,49 @@ H5D__chunk_io_term(const H5D_chunk_map_t *fm)
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5D__chunk_io_term() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5D__chunk_index_close
+ *
+ * Purpose:     Close the given dataset's chunk index, or destroy it if
+ *              `destroy` is true.  A closed index merely releases holds
+ *              on metadata cache entries; the index can be reopened.  Once
+ *              a dataset's index is
+ *              destroyed, however, the dataset must not try to use the
+ *              index, again.
+ *
+ *              A useful side-effect of closing the chunk index is the
+ *              release of pinned/tagged metadata cache entries connected
+ *              with the index.
+ *
+ * Return:      SUCCEED/FAIL
+ *-------------------------------------------------------------------------
+ */
+static herr_t
+H5D__chunk_index_close(const H5D_t *dset, bool destroy)
```

Comment:
This should really be two separate routines, not a Boolean that chooses different function pointers.   Closing an object is definitely different from deleting it.

Reply:
Actually, it’s worse in a sense - the purpose of the 'dest' is to just destroy the in-memory data structure, I.e. close the index.  That's why there's a separate 'delete' callback and all of the chunk index code just copies the 'dest' function pointers.   This routine and the added function pointers should be removed.

Reply:
Perfectly reasonable to also rename the 'dest' callback to 'close', to avoid similar confusion in the future.